packages/language-server/src/__test__/completion.test.ts

```diff
@@ -154,98 +207,121 @@ suite('Completions', function () {
-        { line: 7, character: 10 },
-        {
+      assertCompletion({
+        schema: /* Prisma */ `
+        datasource db {
+            url = |
+        }`,
+        expected: {
           isIncomplete: true,
           items: [quotationMarks, env],
         },
-      ),
-        assertCompletion(
-          dataSourceWithUri,
-          { line: 11, character: 15 },
-          {
-            isIncomplete: false,
-            items: [envArgument],
-          },
-        )
+      })
+      assertCompletion({
+        schema: /* Prisma */ `
+        datasource db {
+            url = env("|")
+        }`,
+        expected: {
+          isIncomplete: false,
+          items: [envArgument],
+        },
+      })
     })
 
     test('Diagnoses single provider suggestions for datasource block', () => {
-      assertCompletion(
-        sqliteDocUri,
-        { line: 14, character: 14 },
-        {
+      assertCompletion({
+        schema: /* Prisma */ `
+        datasource db {
+         provider = "|"
+        }`,
+        expected: {
           isIncomplete: true,
           items: [mysql, postgresql, sqlite, sqlserver, mongodb, cockroachdb],
         },
-      ),
-        assertCompletion(
-          sqliteDocUri,
-          { line: 18, character: 13 },
-          {
-            isIncomplete: true,
-            items: [array, quotationMarks],
-          },
-        )
+      })
+      assertCompletion({
+        schema: /* Prisma */ `
+        datasource db {
+          provider = |
+        }`,
+        expected: {
+          isIncomplete: true,
+          items: [array, quotationMarks],
+        },
+      })
     })
 
+    // TODO: Not valid, we should remove the logic doing this
     test('Diagnoses multiple provider suggestions for datasource block', () => {
-      assertCompletion(
-        sqliteDocUri,
-        { line: 6, character: 15 },
-        {
+      assertCompletion({
+        schema: /* Prisma */ `
+        datasource db {
+          provider = ["|"]
+        }`,
+        expected: {
           isIncomplete: true,
           items: [mysql, postgresql, sqlite, sqlserver, mongodb, cockroachdb],
         },
-      ),
-        assertCompletion(
-          sqliteDocUri,
-          { line: 22, character: 14 },
-          {
-            isIncomplete: true,
-            items: [quotationMarks],
-          },
-        )
-      assertCompletion(
-        sqliteDocUri,
-        { line: 10, character: 25 },
-        {
+      })
+      assertCompletion({
+        schema: /* Prisma */ `
+        datasource db {
+          provider = [|]
+        }`,
+        expected: {
+          isIncomplete: true,
+          items: [quotationMarks],
+        },
+      })
+      assertCompletion({
+        schema: /* Prisma */ `
+        datasource db {
+          provider = ["sqlite", "|"]
+        }`,
+        expected: {
           isIncomplete: true,
           items: [mysql, postgresql, sqlserver, mongodb, cockroachdb],
         },
-      )
+      })
```

Comment:
Is this something that should be worked on in a future PR?

Reply:
Yes, I think it can be a separate one, thanks for reminding me about this!